tests/unit/queries/query.ts

```diff
@@ -951,29 +952,380 @@ export const testQuery = createTest('Query', (ctx) => {
-		const res = await ctx.query({ $entity: 'User', $fields: ['id'] });
-		expect(res).toBeDefined();
-		expect(res).not.toBeInstanceOf(String);
+	// Test retrieving all attributes of an entity using $fields
+	test('should retrieve all attributes of an entity using $fields', async () => {
+		const query = {
+			$entity: 'User',
+			$fields: ['id', 'name', 'email', 'age'],
+		};
+		const result = await runQuery(query);
+		expect(result).toEqual([
+			{
+				id: 'user1',
+				name: 'John Doe',
+				email: 'john@example.com',
+				age: 30,
+			},
+			// ... other users
+		]);
+	});
 
-		expect(deepSort(res)).toEqual([
+	// Test retrieving a subset of attributes using $fields
+	test('should retrieve a subset of attributes using $fields', async () => {
+		const query = {
+			$entity: 'User',
+			$fields: ['id', 'name'],
+		};
+		const result = await runQuery(query);
+		expect(result).toEqual([
 			{
-				$id: 'god1',
-				$thing: 'God',
-				$thingType: 'entity',
-				id: 'god1',
+				id: 'user1',
+				name: 'John Doe',
+			},
+			// ... other users
+		]);
+	});
+
+	// Test retrieving an entity by its unique identifier using $id
+	test('should retrieve an entity by its unique identifier using $id', async () => {
+		const query = {
+			$entity: 'User',
+			$id: 'user1',
+			$fields: ['id', 'name', 'email', 'age'],
+		};
+		const result = await runQuery(query);
+		expect(result).toEqual({
+			id: 'user1',
+			name: 'John Doe',
+			email: 'john@example.com',
+			age: 30,
+		});
+	});
+
+	// Test retrieving multiple entities by their unique identifiers using $id with an array
+	test('should retrieve multiple entities by their unique identifiers using $id with an array', async () => {
+		const query = {
+			$entity: 'User',
+			$id: ['user1', 'user2'],
+			$fields: ['id', 'name', 'email', 'age'],
+		};
+		const result = await runQuery(query);
+		expect(result).toEqual([
+			{
+				id: 'user1',
+				name: 'John Doe',
+				email: 'john@example.com',
+				age: 30,
 			},
 			{
-				$id: 'superuser1',
-				$thing: 'SuperUser',
-				$thingType: 'entity',
-				id: 'superuser1',
+				id: 'user2',
+				name: 'Jane Smith',
+				email: 'jane@example.com',
+				age: 25,
+			},
+		]);
+	});
+
+	// Test retrieving entities based on a specific attribute value using $filter
+	test('should retrieve entities based on a specific attribute value using $filter', async () => {
+		const query = {
+			$entity: 'User',
+			$filter: {
+				age: 30,
+			},
+			$fields: ['id', 'name', 'email', 'age'],
+		};
+		const result = await runQuery(query);
+		expect(result).toEqual([
+			{
+				id: 'user1',
+				name: 'John Doe',
+				email: 'john@example.com',
+				age: 30,
+			},
+		]);
+	});
+
+	// Test retrieving entities based on multiple attribute values using $filter with $and
+	test('should retrieve entities based on multiple attribute values using $filter with $and', async () => {
+		const query = {
+			$entity: 'User',
+			$filter: {
+				$and: [{ age: 30 }, { name: 'John Doe' }],
```

Comment:
This is not needed like this, because this works 
$filter: {age:30, name: John Doe} as a AND